packages/react-ui-core/src/Menu/Menu.js

```diff
@@ -93,28 +93,31 @@ export default class Menu extends PureComponent {
     }
   }
 
-  get value() {
-    return this.options[this.state.highlightIndex]
-  }
-
   get options() {
     return this.props.options || []
   }
 
   get items() {
-    return this.options.map(option => (
-      typeof option === 'object' ? option.label : option
-    ))
+    return this.options.map(option =>
+      ((typeof option === 'object' && option.label !== undefined) ?
+        option.label :
+        option
+      )
+    )
+  }
+
+  get highlightedOption() {
+    return this.options[this.state.highlightIndex || 0]
   }
 
   @autobind
   handleSelection() {
-    const { onItemSelect, options } = this.props
+    const { onItemSelect } = this.props
 
-    const option = options[this.state.highlightIndex]
+    const value = this.highlightedOption
```

Comment:
I'm still confused why this is `value` rather than `option`. If it is named `value` then we end up with an object named `value` that has a `value` attribute